### DISCUSSION TITLE
problemas para subir mis kastas desde la git bash

### DISCUSSION THREAD

**Discussion by Alcomandez** [LaunchX-InnovaccionVirtual/CursoIntroPython#303] [Q&A]
alguien sabe porque me sale este error y no me deja subir mis katas? , ya lo intente desde la cmd y me sale el mismo error 
![image](https://user-images.githubusercontent.com/89067301/154611054-e142bfd9-7ea0-4480-af55-fe08b44aa150.png)

**Comment by NayGL** [comment]
Hola, todo esto es nuevo para mí, pero en el último vídeo viene tambíen que podemos subir los archivos así:

![upload Katas](https://user-images.githubusercontent.com/94994465/154611948-3e2b6e0b-4820-4f0f-b2cd-26250c406ca4.png)

**Reply by Alcomandez** [reply]
Muchas gracias, me hiciste la noche 💯

**Comment by ArielCanedo** [comment]
Puedes arrastrar los archivos en la página de GitHub

**Reply by Alcomandez** [reply]
Muchas gracias funciono igual :D

**Comment by JavierMendez-Coder** [comment]
Buenas noches. Sé que tu problema ya fue solucionado, pero igual deseo ayudarte respecto a tu problema con GitBash/Terminal.
Lo que sucede es que aún no estas agregando nada a tu `staging area`, y eso lo puedes observar gracias al comando:
```git
$ git status
```
Este comando te muestra de verde (en GitBash) los cambios realizados al `staging area`. Por eso puedes observar que todos tus archivos aparecen de color rojo.
Para agregarlos al `staging area` puedes utilizar el comando:
```
$ git add <archivo>...
```
Ejemplos: 
1. Para un solo archivo: `$ git add K0.ipynb`
2. Para dos o más archivos: `$ git add K1.ipynb K2.ipynb`
3. Para todos los archivos: `$ git add .`

Además, tambien tienes que hacer `commit` de los cambios del `staging area` al repositorio local, con el comando:
```
$ git commit -m "Título del commit" -m "Descripción del commit"
```
También, para establecer el repositorio remoto puedes realizarlo de la siguiente forma:
```
$ git remote add <nombre_del_repositorio> <url_del_repositorio>
```
Ejemplo: `$ git remote add origin https://github.com/Alcomandez/katas.git` 
En este caso el repositorio remoto se llama `origin` pero puede ser el nombre que desees y el url es el enlace a tu repositorio de GitHub con `.git` al final.
<br>
Finalmente, para subir tus cambios se utiliza este comando:
```
$ git push <nombre_del_repositorio> <rama_del_repositorio>
```
Ejemplo: `$ git push origin master` 
Siguiendo con el ejemplo anterior usamos `origin` como nombre del repositorio (al igual que antes, es el nombre que tu desees), y con el nombre de la rama al que quieres que se suban, en este caso `master`.

Bueno eso es todo, igual te recomiendo buscar un poco más por tu parte por si cometí un error. Saludos. 👋

¡Una cosa más! en caso de que alguien responda tu pregunta, por favor archiva la discución o marca una sugerencia como respuesta. Gracias.

**Comment by EnriqueValderrama** [comment]
Hola, se que la pregunta ya está más que respondida, pero si a alguien más le cuesta usar la terminal como a mi, pueden seguir este tutorial de como hacerlo directamente desde Visual Studio 
https://youtu.be/yhlArNbzWgE

**Reply by Gracesl** [reply]
gracias
